### PR TITLE
WooCommerce: Sidebar animation

### PR DESCRIPTION
**Pull request by kellychoffman**
This addresses https://github.com/Automattic/wp-calypso/pull/13837#issuecomment-300624125.

To test: 

- Navigate to http://calypso.localhost:3000/store/{site}
- Confirm that sidebar sliiiiides in

@jameskoster: What do you think? We might have to wait until the Store menu item is implemented to really see how it feels.

### CODE REVIEW

**Comment by belcherj** [comment]
The code looks good to me.  It does seem odd though that the skeleton has the left menu and then when the site finally loads it slides in.  Should this not a be a calypso wide setting not just the WooCommerce plugin?

**Comment by jameskoster** [comment]
Applied the animation style from the editor for consistency.

**Comment by kellychoffman** [comment]
> 1 second feels a little slow, I might riff on that in a separate PR xD

Riff on it here! this is what this PR is for.

> I also think the animation should only occur when you open / close the nav rather than every page load.

Good call. I'm not sure how to do that, though. Do you? Or perhaps @belcherj?

> It does seem odd though that the skeleton has the left menu and then when the site finally loads it slides in. Should this not a be a calypso wide setting not just the WooCommerce plugin?

Could you clarify - which part should be the calypso wide setting?

**Comment by belcherj** [comment]
> Could you clarify - which part should be the calypso wide setting?

There are other sidebars in Calypso.  Shouldn't we be using the same style and format as the rest of the site?

>I also think the animation should only occur when you open/close the nav rather than every page load. Good call. I'm not sure how to do that, though. Do you? Or perhaps @belcherj?

It would take some looking not sure how that would work.

**Comment by kellychoffman** [comment]
> There are other sidebars in Calypso. Shouldn't we be using the same style and format as the rest of the site?

Yep, but just like the editor sidebar, this is unique to the site sidebar in that it replaces and overlays it. We are already sharing the style and layout of the site sidebar, just the animation is different.

**Comment by jameskoster** [comment]
> Good call. I'm not sure how to do that, though. Do you? 

If there's no actual navigation/reload when the 'Store' menu item (or the <-) is clicked then I think all we'd need to do is toggle a visibility css class on the store sidebar. Then animate depending on the class.

File: client/extensions/woocommerce/style.scss [hunk 1]
```diff
@@ -12,8 +12,18 @@
 	@import 'components/form-dimensions-input/style';
 }
 
-.store-sidebar__sidebar {
+@keyframes slideInFromLeft {
+	0% {
+		transform: translateX(-273px);
```
Comment (justinshreve) [inline client/extensions/woocommerce/style.scss]: There needs to be [spacing around parenthesis](https://github.com/Automattic/wp-calypso/blob/de700a238a201232c8481b59b4f7feba7700e977/docs/coding-guidelines/css.md#spacing).

File: client/extensions/woocommerce/style.scss [hunk 2]
```diff
@@ -12,8 +12,18 @@
...
+	}
+	100% {
+		transform: translateX(0);
```
Comment (justinshreve) [inline client/extensions/woocommerce/style.scss]: Spacing.

File: client/extensions/woocommerce/style.scss [hunk 3]
```diff
@@ -12,8 +12,18 @@
...
+	}
+}
+
+.sidebar.store-sidebar__sidebar {
 	display: block;
+	animation: 0.15s cubic-bezier(0.075, 0.82, 0.165, 1) 0s 1 slideInFromLeft;
```
Comment (justinshreve) [inline client/extensions/woocommerce/style.scss]: Spacing.

**Comment by justinshreve** [comment]
I definitely agree we need to see it in action because it does indeed feel incomplete / a bit jarring right now on it's own.

**Comment by kellychoffman** [comment]
Agree. I'm going to close this for now. It was an experiment and we can try again when the Store nav item is implemented. Thanks for the review!

**Pull request closed without merging**